### PR TITLE
hdfs: add resources allocation management

### LINKED ISSUE #556: hdfs: manage resources allocation

### PR DESCRIPTION
**Pull request by mehdibn**
#### Which issue(s) this PR fixes

Fixes #556 

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

**Review by rpignolet — APPROVED** [review]
For me it is weird to force the `Xms` and `Xmx` to the same value, it will be too complex to have a variable for `Xms` and another for `Xmx` ?

**Comment by mehdibn** [comment]
> For me it is weird to force the `Xms` and `Xmx` to the same value, it will be too complex to have a variable for `Xms` and another for `Xmx` ?

yes, we can do as suggested but the idea was to force this good practice described above: 


When a JVM process starts, it initially allocates the minimum heap size. If the need for memory arises at runtime, it can go back to the OS and allocate more, up to the maximum heap size. 

**Why do we recommend setting these to the same value?** 

Those incremental memory allocations can cause unpredictable runtime performance. Additionally, starting with a low minimum heap size could cause instability as the need for a larger heap grows over time. If it needs to grow to the maximum, but the OS doesn’t actually have that amount of memory available to hand out, then the process could be subject to OutOfMemoryError conditions, or possibly even the [OOM killer](http://linux-mm.org/OOM_Killer) stopping the whole process on Linux. Setting minimum and maximum to the same value eliminates these potential sources of instability.

File: tdp_vars_defaults/hdfs/hdfs.yml [hunk 1]
```diff
@@ -84,3 +84,13 @@ hdfs_check_retries: 0
 hdfs_check_delay: 5
 hdfs_check_port_str: "Connected to"
 hdfs_check_path_file: "/tmp/hdfs_service_check_{{ ansible_date_time.iso8601_basic }}.file"
+
+# HDFS resources allocation
+hdfs_namenode_heapsize: 2048m
```
Comment (nschung) [inline tdp_vars_defaults/hdfs/hdfs.yml:89]: Are they the default values?

Comment (mehdibn) [reply tdp_vars_defaults/hdfs/hdfs.yml:89]: all default values are `1024m`, i increased exceptionally the namenode one.

Comment (leopaul36) [reply tdp_vars_defaults/hdfs/hdfs.yml:89]: I don't think there is a default value. I think it depends on the Java version.

With Java 1.8.0_272 I have Xms = 130023424 and Xmx = 2051014656

**Review by leopaul36 — APPROVED** [review]

File: tdp_vars_defaults/hdfs/hdfs.yml (hunk 1 above)
Comment (mehdibn) [reply tdp_vars_defaults/hdfs/hdfs.yml:89]: @leopaul36 @nschung the default values that i mentioned are the HDP ones

**Review by nschung — APPROVED** [review]

**Pull request merged**